test/micro/org/openjdk/bench/java/lang/StringDecode.java

```diff
@@ -70,90 +62,125 @@ public void setup() {
-             dignissim diam eu ultricies. Class aptent taciti sociosqu ad litora torquent
-             per conubia nostra, per inceptos himenaeos. Suspendisse erat diam, fringilla
-             sed massa sed, posuere viverra orci. Suspendisse tempor libero non gravida
-             efficitur. Vivamus lacinia risus non orci viverra, at consectetur odio laoreet.
-             Suspendisse potenti.
-
-             Phasellus vel nisi iaculis, accumsan quam sed, bibendum eros. Sed venenatis
-             nulla tortor, et eleifend urna sodales id. Nullam tempus ac metus sit amet
-             sollicitudin. Nam sed ex diam. Praesent vitae eros et neque condimentum
-             consectetur eget non tortor. Praesent bibendum vel felis nec dignissim.
-             Maecenas a enim diam. Suspendisse quis ligula at nisi accumsan lacinia id
-             hendrerit sapien. Donec aliquam mattis lectus eu ultrices. Duis eu nisl
-             euismod, blandit mauris vel, placerat urna. Etiam malesuada enim purus,
-             tristique mollis odio blandit quis. Vivamus posuere.
-            """.getBytes(charset);
-
-        longLatin1String = """
-             a\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             b\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             c\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             d\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             e\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             f\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             g\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             h\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             i\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6
-             j\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6\u00F6
-             k\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6\u00F6
-             l\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6\u00F6
-             m\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00F6\u00B6\u00F6\u00F6
-            """.getBytes(charset);
+             Suspendisse potenti.""";
+    private static final String UTF16_STRING = "\uFF11".repeat(31);
+    private static final String LATIN1_STRING = "\u00B6".repeat(31);
+
+    @Setup
+    public void setup() {
+        charset = Charset.forName(charsetName);
+        asciiString = LOREM.substring(0, 32).getBytes(charset);
```

Comment:
This is problematic IMO in that it's missing short strings such as "Claes". Average Java strings are about 32 bytes long AFAICR, and people writing (vectorized) ijntrinsics have a nasty habit of optimizing for long strings, to the detriment of typical-length ones.
Whether we like it or not, people will optimize for benchmarks, so it's important that benchmark data is realistic. The shortest here is 15 bytes, as far as I can see. I'd certainly include a short string of just a few bytes so that intrinsics don't cause regressions in important cases.

Reply:
All good points. I've added a number of such short variants to all(?) relevant microbenchmarks. The tests should now better cover a mix of input lengths and encodings.

Reply:
👍